### PR TITLE
Collage, Container, Datapoint, Fieldset, Heading: use generated props tables

### PR DESCRIPTION
**Pull request by dangerismycat**
Refactor docs to use generated props tables: Collage, Container, Datapoint, Fieldset, Heading

Also, removed unneeded React imports (probably from an older version of the scaffolding script)

### CODE REVIEW

File: packages/gestalt/src/Heading.js [hunk 1]
```diff
@@ -4,39 +4,81 @@ import cx from 'classnames';
...
+  /**
+   * Visually truncate the text to the specified number of lines. This also adds the `title` attribute if `children` is a string, which displays the full text on hover in most browsers. See [example below](#overflowTruncation) for more details.
+   */
   lineClamp?: number,
+  /**
+   * How truncation is handled when text overflows the line. See [example below](#overflowTruncation) for more details.
+   */
   overflow?: Overflow,
+  /**
+   * The font size of the text. See [example below](#sizes) for more details.
```
Comment (christianvuerings) [inline packages/gestalt/src/Heading.js:76]: @dangerismycat could we make sure to include `https://gestalt.pinterest.systems/` in all the URLs? That way these links will also work in our flow docs:

```
The font size of the text. See [example](https://gestalt.pinterest.systems/heading#sizes) for more details.
```

Comment (christianvuerings) [reply packages/gestalt/src/Heading.js:76]: Also, should we change `See example below` to `See example` - the "below" portion doesn't make a lot of sense in flow docs.

**Comment by AlbertCarreras** [comment]
Screen Shot 2022-01-06 at 1 49 37 PM

**Review by AlbertCarreras — APPROVED** [review]
LGTM! @dangerismycat Thanks for these changes. Agree with Christian re absolute URLs + removing "below". R+ing so you can merge when you fix them.

File: packages/gestalt/src/Heading.js (hunk 1 above)
Comment (dangerismycat) [reply packages/gestalt/src/Heading.js:76]: Ah good point, still clearing the cobwebs from my holiday brain 😅

**Pull request merged**